src/bundler.ts

```diff
@@ -118,62 +116,138 @@ class Bundler {
+    const scriptContent = backingScript.document.contents;
+    dom5.removeAttribute(externalScript, 'src');
+    dom5.setTextContent(externalScript, scriptContent);
   }
 
   /**
-   * Replace htmlImport
+   * Inline external stylesheets <link type="text/css" href="*">
    */
-  async inlineImport(
-      documentUrl: string, htmlImport: ASTNode, analyzer: Analyzer,
+  async inlineCss(documentUrl: string, externalStylesheet: ASTNode):
+      Promise<void> {
+    const rawUrl: string = dom5.getAttribute(externalStylesheet, 'href')!;
+    const resolved = url.resolve(documentUrl, rawUrl);
+    const backingStylesheet: ScannedDocument =
+        await this.analyzer._scanResolved(resolved);
+    const stylesheetContent = backingStylesheet.document.contents;
+    dom5.removeAttribute(externalStylesheet, 'href');
+    dom5.setTextContent(externalStylesheet, stylesheetContent);
+  }
+
+  /**
+   * Inline external HTML files <link type="import" href="*">
+   * TODO(usergenic): Refactor method to simplify and encapsulate case handling
+   *     for hidden div adjacency etc.
+   */
+  async inlineHtmlImport(
+      documentUrl: string, htmlImport: ASTNode,
       inlined: Set<string>): Promise<void> {
-    const rawUrl: string = dom5.getAttribute(htmlImport, 'href');
+    const rawUrl: string = dom5.getAttribute(htmlImport, 'href')!;
     const resolved = url.resolve(documentUrl, rawUrl);
     if (inlined.has(resolved)) {
       dom5.remove(htmlImport);
       return;
     }
     inlined.add(resolved);
     const backingDocument: ScannedDocument =
-        await analyzer._analyzeResolved(resolved);
+        await this.analyzer._scanResolved(resolved);
     const documentAst = dom5.parseFragment(backingDocument.document.contents);
     this.pathResolver.resolvePaths(documentAst, resolved, documentUrl);
-    const importParent = htmlImport.parentNode;
+    let importParent: ASTNode;
+    if (matchers.afterHiddenDiv(htmlImport)) {
+      importParent = dom5.nodeWalkPrior(htmlImport, matchers.hiddenDiv)!;
+      dom5.remove(htmlImport);
+      dom5.append(importParent, htmlImport);
+    } else if (matchers.beforeHiddenDiv(htmlImport)) {
+      const index = htmlImport.parentNode!.childNodes!.indexOf(htmlImport);
+      importParent = htmlImport.parentNode!.childNodes![index + 1];
+      dom5.remove(htmlImport);
+      ASTUtils.prepend(importParent, htmlImport);
+    } else if (!matchers.inHiddenDiv(htmlImport)) {
+      const hiddenDiv = this.getHiddenNode();
+      dom5.replace(htmlImport, hiddenDiv);
+      dom5.append(hiddenDiv, htmlImport);
+      importParent = hiddenDiv;
+    } else {
+      importParent = htmlImport.parentNode!;
+    }
+
+    ASTUtils.insertAllBefore(importParent, htmlImport, documentAst.childNodes!);
     dom5.remove(htmlImport);
-    ASTUtils.prependMultiple(importParent, documentAst.childNodes);
   }
 
+  /**
+   * Given a URL to an entry-point html document, produce a single document
+   * suitable for deployment.
```

Comment:
"suitable for deployment" describes what the result might be used for, but not what the method does. "with HTML imports, external stylesheets and external scripts inlined, according to the options for this Bundler"?